python/veles/tests/db/test_db.py

```diff
@@ -0,0 +1,342 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import unicode_literals
+
+import unittest
+import tempfile
+import os.path
+
+import six
+
+from veles.db import Database
+from veles.data.bindata import BinData
+from veles.proto.node import Node
+from veles.schema.nodeid import NodeID
+
+
+class TestDatabase(unittest.TestCase):
+    def test_simple(self):
+        db = Database(None)
+        node = Node(id=NodeID(),
+                    tags={'my_tag'},
+                    attr={'my_attr': 'my_val'},
+                    pos_start=0x123,
+                    pos_end=0x456,
+                    data={'my_key'},
+                    bindata={'my_bindata': 12345})
+        db.create(node)
+        n1 = db.get(NodeID())
+        self.assertEqual(n1, None)
+        n2 = db.get(node.id)
+        self.assertIsInstance(n2, Node)
+        self.assertEqual(n2.id, node.id)
+        self.assertEqual(n2.parent, node.parent)
+        self.assertEqual(n2.tags, node.tags)
+        self.assertEqual(n2.attr, node.attr)
+        self.assertEqual(n2.pos_start, node.pos_start)
+        self.assertEqual(n2.pos_end, node.pos_end)
+        self.assertEqual(n2.data, set())
+        self.assertEqual(n2.bindata, {})
+        with self.assertRaises(TypeError):
+            db.get('zlew')
+        with self.assertRaises(TypeError):
+            db.get(node.id.bytes)
+        with self.assertRaises(TypeError):
+            db.create({'id': node.id})
+
+    def test_persist(self):
+        d = tempfile.mkdtemp()
+        path = os.path.join(d, 'x.db')
+        db1 = Database(path)
+        node = Node(id=NodeID(),
+                    tags={'my_tag'},
+                    attr={'my_attr': 'my_val'},
+                    pos_start=0x123,
+                    pos_end=0x456,
+                    data={'my_key'},
+                    bindata={'my_bindata': 12345})
+        db1.create(node)
+        db1.close()
+        db2 = Database(path)
+        n1 = db2.get(NodeID())
+        self.assertEqual(n1, None)
+        n2 = db2.get(node.id)
+        self.assertIsInstance(n2, Node)
+        self.assertEqual(n2.id, node.id)
+        self.assertEqual(n2.parent, node.parent)
+        self.assertEqual(n2.tags, node.tags)
+        self.assertEqual(n2.attr, node.attr)
+        self.assertEqual(n2.pos_start, node.pos_start)
+        self.assertEqual(n2.pos_end, node.pos_end)
+        self.assertEqual(n2.data, set())
+        self.assertEqual(n2.bindata, {})
+        db2.close()
+        os.unlink(path)
+        os.rmdir(d)
```

Comment:
On windows files in temp never get deleted automatically, so I'd change it so that we try to clean-up before failing the test to avoid unnecessary additional garbage.